src/main/tables.ts

```diff
@@ -32,45 +25,31 @@ async function syncTables(): Promise<void> {
 // ----------------------------------------------------------------
 
 export async function loadTables(): Promise<ImportedTable[]> {
-  const client = new pg.Client(connectionConfig);
-  await client.connect();
-
   const { adminUser } = postgresConfig;
-  try {
-    const ret: ImportedTable[] = [];
-    const allTables = await client.query(`SELECT tablename FROM pg_catalog.pg_tables WHERE tableowner='${adminUser}';`);
-    allTables.rows.forEach((row) => ret.push({ key: row.tablename, name: row.tablename, aidColumns: [] }));
-
-    const aids = await client.query(
-      'SELECT tablename, objname FROM pg_catalog.pg_tables, diffix.show_labels() ' +
-        "WHERE objname ~ ('public\\.\\\"?' || tablename || '\\\"?\\..*') AND" +
-        `      tableowner='${adminUser}' AND ` +
-        "      label='aid';",
-    );
-
-    aids.rows.forEach((row) =>
-      ret.find(({ name }) => name == row.tablename)?.aidColumns.push(row.objname.split('.').at(-1)),
-    );
-
-    const tables = ret.map((table) => `'${table.name}'`).join(', ');
-    console.info(`Found the following tables: ${tables}.`);
-
-    return ret;
-  } finally {
-    client.end();
-  }
+
+  const allTables = await sql`SELECT tablename FROM pg_catalog.pg_tables WHERE tableowner = ${adminUser}`;
+
+  const ret: ImportedTable[] = allTables.map((row) => ({ key: row.tablename, name: row.tablename, aidColumns: [] }));
+
+  const aids = await sql`
+    SELECT tablename, objname
+    FROM pg_catalog.pg_tables, diffix.show_labels()
+    WHERE objname ~ ('public\\.\\\"?' || tablename || '\\\"?\\..*')
+      AND tableowner = ${adminUser}
+      AND label = 'aid'
+  `;
+
+  aids.forEach((row) => find(ret, { name: row.tablename })?.aidColumns.push(row.objname.split('.').at(-1)));
+
+  const tables = ret.map((table) => `'${table.name}'`).join(', ');
+  console.info(`Found the following tables: ${tables}.`);
+
+  return ret;
 }
 
 export async function removeTable(tableName: string): Promise<void> {
-  const client = new pg.Client(connectionConfig);
-  await client.connect();
-
-  try {
-    await client.query(`DROP TABLE public."${tableName}";`);
-    await syncTables();
-  } finally {
-    client.end();
-  }
+  await sql`DROP TABLE public.${sql(tableName)}`;
```

Comment:
I presume the `sql` handles correct quoting?

Reply:
Yes, `${someString}` adds single quotes, and `${sql(someString)}` adds double quotes. A bit too much magic and takes getting used to.